crates/ide/src/hover.rs

```diff
@@ -111,16 +111,44 @@ pub(crate) fn hover(
         kind if kind.is_trivia() => 0,
         _ => 1,
     })?;
-    let token = sema.descend_into_macros(token);
-    let node = token.parent()?;
+
+    sema.descend_into_macros_many(token)
+        .iter()
+        .filter_map(|token| match token.parent() {
+            Some(node) => find_hover_result(db, &sema, file_id, offset, config, token, &node),
+            None => None,
+        })
+        // reduce all descends into a `RangeInfo` that spans from the earliest start to the latest end,
+        // concatenates all `Markup`s with `\n`,
+        // and accumulates all actions into the `actions` vector.
+        .reduce(|mut acc, RangeInfo { range, mut info }| {
+            let start = acc.range.start().min(range.start());
+            let end = acc.range.end().max(range.end());
+
+            acc.range = TextRange::new(start, end);
+            acc.info.actions.append(&mut info.actions);
+            acc.info.markup = Markup::from(format!("{}\n{}", acc.info.markup, info.markup));
```

Comment:
```suggestion
            acc.info.markup = Markup::from(format!("{}\n---\n{}", acc.info.markup, info.markup));
```
probably better to read with a divider inbetween